src/generate/context.rs

```diff
@@ -228,25 +200,92 @@ impl CppContext {
         }
 
         println!("Writing {:?}", self.typedef_path.as_path());
-        let mut writer = CppWriter {
+        let mut typedef_writer = CppWriter {
             stream: File::create(self.typedef_path.as_path())?,
             indent: 0,
         };
+        let _typeimpl_writer = CppWriter {
+            stream: File::create(self.type_impl_path.as_path())?,
+            indent: 0,
+        };
+        let _fundamental_writer = CppWriter {
+            stream: File::create(self.fundamental_path.as_path())?,
+            indent: 0,
+        };
+
         // Write includes for typedef
         self.typedef_includes
             .iter()
-            .for_each(|inc| inc.write(&mut writer).unwrap());
-        self.declarations
-            .iter()
-            .for_each(|dec| dec.write(&mut writer).unwrap());
-        self.types.iter().for_each(|(k, v)| {
-            writeln!(writer, "/* {:?} */", k).unwrap();
-            v.write(&mut writer).unwrap();
+            .for_each(|inc| inc.write(&mut typedef_writer).unwrap());
+
+        // forward declares
+        for (namespace, strings) in &self.typedef_declarations {
+            writeln!(typedef_writer, "namespace {} {{", namespace)?;
+            strings
+                .iter()
+                .for_each(|s| s.write(&mut typedef_writer).unwrap());
+            writeln!(typedef_writer, "}} // namespace {}", namespace)?;
+        }
+        self.typedef_types.iter().for_each(|(k, v)| {

```

Comment:
Typedef types should be present in the typedef declarations, ideally it should fall under a particular namespace. This is not a big deal though.